coil-base/src/main/java/coil/decode/ImageSource.kt

```diff
@@ -133,14 +141,18 @@ sealed class ImageSource : Closeable {
      * Return a [File] containing this [ImageSource]'s data.
      * If this image source is backed by a [BufferedSource], a temporary file containing this
      * [ImageSource]'s data will be created.
+     *
+     * This method is only safe when used on FileSystem.SYSTEM.
```

Comment:
I think this is always safe, right? We only need to create a temp file if the `ImageSource` is backed by a `BufferedSource` in which case we should always use `FileSystem.SYSTEM` to create the temp file.